dev/com.ibm.ws.transaction.context/src/com/ibm/ws/transaction/context/internal/SerialTransactionContextImpl.java

```diff
@@ -0,0 +1,158 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.transaction.context.internal;
+
+import java.io.IOException;
+import java.io.NotSerializableException;
+import java.io.ObjectOutputStream;
+import java.util.concurrent.RejectedExecutionException;
+
+import javax.transaction.InvalidTransactionException;
+import javax.transaction.SystemException;
+import javax.transaction.Transaction;
+
+import com.ibm.tx.jta.embeddable.EmbeddableTransactionManagerFactory;
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.ws.LocalTransaction.LocalTransactionCoordinator;
+import com.ibm.ws.LocalTransaction.LocalTransactionCurrent;
+import com.ibm.ws.Transaction.UOWCurrent;
+import com.ibm.ws.tx.embeddable.EmbeddableWebSphereTransactionManager;
+import com.ibm.ws.uow.embeddable.UOWManager;
+import com.ibm.ws.uow.embeddable.UOWManagerFactory;
+import com.ibm.ws.uow.embeddable.UOWToken;
+import com.ibm.wsspi.threadcontext.ThreadContext;
+
+/**
+ * A special transaction context implementation for MicroProfile Concurrency that propagates a
+ * transaction to another thread on the condition that it is only active on one thread at a time.
+ */
```

Comment:
Should we add a TODO to potentially update this wording if the testTwoThreadsConcurrentlyOperateInTransaction() test turns out to be valid?

Reply:
This is temporary experimental code. The whole thing will be revised if we actually decide to support it.